demos/security_barrier_camera_demo/main.cpp

```diff
@@ -333,26 +338,36 @@ void Drawer::process() {
         lastShownframeId++;
         cv::Mat mat = firstGridIt->second.getMat();
 
-        float opacity = 0.6f;
-        fillROIColor(mat, cv::Rect(5, 5, 700, 115), cv::Scalar(255, 0, 0), opacity);
-
-        std::ostringstream out;
-        out << "Mean overall time per all inputs: " << std::fixed << std::setprecision(2) << std::setw(6);
-        const auto t1 = std::chrono::steady_clock::now();
-        uint64_t frameCounter = context.frameCounter;
-        const ms meanOverallTimePerAllInputs = std::chrono::duration_cast<ms>((t1 - context.t0)
-                                               * context.readersContext.inputChannels.size()) / frameCounter;
-        out << meanOverallTimePerAllInputs.count();
-        out << "ms /" << std::setw(6) << std::chrono::seconds(1) / meanOverallTimePerAllInputs << "FPS";
-
-        cv::putText(mat, out.str(), cv::Point2f(10, 35), cv::FONT_HERSHEY_TRIPLEX, 0.7, cv::Scalar{255, 255, 255});
-        cv::putText(mat, "Detection InferRequests usage", cv::Point2f(10, 70), cv::FONT_HERSHEY_TRIPLEX, 0.7, cv::Scalar{255, 255, 255});
-        cv::Rect usage(15, 90, 400, 20);
+        constexpr float OPACITY = 0.6f;
+        fillROIColor(mat, cv::Rect(5, 5, 390, 115), cv::Scalar(255, 0, 0), OPACITY);
+        cv::putText(mat, "Detection InferRequests usage", cv::Point2f(15, 70), cv::FONT_HERSHEY_TRIPLEX, 0.7, cv::Scalar{255, 255, 255});
+        cv::Rect usage(15, 90, 370, 20);
         cv::rectangle(mat, usage, {0, 255, 0}, 2);
         uint64_t nireq = context.nireq;
+        uint64_t frameCounter = context.frameCounter;
         usage.width = static_cast<int>(usage.width * static_cast<float>(frameCounter * nireq - context.freeDetectionInfersCount) / (frameCounter * nireq));
         cv::rectangle(mat, usage, {0, 255, 0}, cv::FILLED);
 
+        if (std::chrono::steady_clock::time_point() == context.drawersContext.localT0) {
```

Comment:
This logic is dubious, because it assumes that the clock's epoch can never be returned by `now()`. This might not be true, because `steady_clock` is an abstract clock, and it might return negative values at the start.

You should just initialize `localT0` at the same time you initialize `t0`.

Reply:
Updated